dev/tasks/generate-crds

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env bash
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+set -x
+
+REPO_ROOT="$(git rev-parse --show-toplevel)"
+cd ${REPO_ROOT}
+
+# Run controller-gen to generate CRDs
+cd ${REPO_ROOT}/apis
+go run sigs.k8s.io/controller-tools/cmd/controller-gen@v0.14.0 object crd:crdVersions=v1,allowDangerousTypes=true output:crd:artifacts:config=config/crd/ paths="./..."
+#go run k8s.io/code-generator/cmd/deepcopy-gen@v0.29.0 -O zz_generated.deepcopy -h ../hack/boilerplate.go.txt -i  github.com/GoogleCloudPlatform/k8s-config-connector/apis/... --trim-path-prefix github.com/GoogleCloudPlatform/k8s-config-connector/apis
+
+# Add OpenAPI validation to refs
+cd ${REPO_ROOT}
+go run ./scripts/add-validation-to-crds --dir apis/config/crd/
+
+# Apply a few small tweaks to the CRDs
+cd ${REPO_ROOT}
+go run ./scripts/crd-tools set-field spec.preserveUnknownFields=false --dir apis/config/crd/
+go run ./scripts/crd-tools delete-field status --dir apis/config/crd/
+go run ./scripts/crd-tools set-annotation cnrm.cloud.google.com/version=0.0.0-dev --dir apis/config/crd/
+go run ./scripts/crd-tools delete-annotation controller-gen.kubebuilder.io/version --dir apis/config/crd/
```

Comment:
just curious, any of this can be achieved by kustomize?

Reply:
Maybe, but I think we'd have to use KRM functions (e.g. delete-annotation).  I figure we can start this way and then hopefully create KRM functions.